types/ramda/tools.d.ts

```diff
@@ -187,6 +187,174 @@ export type KeyValuePair<K, V> = [K, V];
  */
 export type Lens<S, A> = (functorFactory: (a: A) => Functor<A>) => (s: S) => Functor<S>;
 
+/*
+ * Returns true if T1 array length less than or equal to length of array T2, else returns false
+ *
+ * @param T1 - first readonly array
+ * @param T2 - second readonly array
+ *
+ * <created by @valerii15298>
+ * */
+type arr1LessThanOrEqual<
+    T1 extends ReadonlyArray<any>,
+    T2 extends ReadonlyArray<any>,
+> = T1['length'] extends T2['length']
+    ? true
+    : T2['length'] extends 0
+    ? false
+    : T2 extends readonly [infer First, ...infer Rest]
+    ? arr1LessThanOrEqual<T1, Rest>
+    : never;
```

Comment:
This is already in the `tools.d.ts`

Reply:
Line 198 in `tools.d.ts` you added the same type in your first commit

Reply:
There is only one such generic added in the first commit, line 198.
I do not see any other generic with the same functionality already included in `tools.d.ts`.
I thought you meant that there is another one in terms of functionality already included in `tools.d.ts` but I do not see another similar generic.

Reply:
Sorry, I'm an idiot. I was reviewing this on my phone.